subsys/bluetooth/mesh/gatt.h

```diff
@@ -12,8 +14,8 @@
 	.interval_min = BT_GAP_ADV_FAST_INT_MIN_2,                             \
 	.interval_max = BT_GAP_ADV_FAST_INT_MAX_2
 
-#define BT_DEVICE_NAME (IS_ENABLED(CONFIG_BT_DEVICE_NAME_DYNAMIC) ? \
-			(const uint8_t *)bt_get_name() : \
-			(const uint8_t *)CONFIG_BT_DEVICE_NAME)
-#define BT_DEVICE_NAME_LEN (IS_ENABLED(CONFIG_BT_DEVICE_NAME_DYNAMIC) ? strlen(bt_get_name()) : \
-			    (sizeof(CONFIG_BT_DEVICE_NAME) - 1))
+#if defined(CONFIG_BT_GAP_DEVICE_NAME_DYNAMIC)
+#define DEVICE_NAME_SIZE CONFIG_BT_GAP_DEVICE_NAME_DYNAMIC_MAX
+#else
+#define DEVICE_NAME_SIZE BT_GAP_DEVICE_NAME_MAX_SIZE
```

Comment:
Since we change this API, do we really need split between dynamic and not-dynamic name lengths? Can't we use one as a Kconfig option limiting its range through `range` option?